src/agent/advertising_proxy.cpp

```diff
@@ -144,79 +144,14 @@ void AdvertisingProxy::AdvertisingHandler(otSrpServerServiceUpdateId aId,
     // incremental ID to match oustanding SRP updates.
     OTBR_UNUSED_VARIABLE(aTimeout);
 
-    otbrError                 error = OTBR_ERROR_NONE;
-    const char *              fullHostName;
-    std::string               hostName;
-    std::string               hostDomain;
-    const otIp6Address *      hostAddress;
-    uint8_t                   hostAddressNum;
-    bool                      hostDeleted;
-    const otSrpServerService *service;
-    OutstandingUpdate *       update;
-
-    mOutstandingUpdates.resize(mOutstandingUpdates.size() + 1);
-    update = &mOutstandingUpdates.back();
-
-    fullHostName = otSrpServerHostGetFullName(aHost);
-
-    otbrLogInfo("Advertise SRP service updates: host=%s", fullHostName);
-
-    SuccessOrExit(error = SplitFullHostName(fullHostName, hostName, hostDomain));
-    hostAddress = otSrpServerHostGetAddresses(aHost, &hostAddressNum);
-    hostDeleted = otSrpServerHostIsDeleted(aHost);
+    OutstandingUpdate *update;
+    otbrError          error = OTBR_ERROR_NONE;
 
+    mOutstandingUpdates.emplace_back();
+    update      = &mOutstandingUpdates.back();
```

Comment:
I prefer adding/popping the new OutstandingUpdate to the vector in `PublishHostAndItsServices` and pass out the newly added entry back with the `OutstandingUpdate *aUpdate` parameter. Thoughts?

Reply:
Does that mean the method signature will be `otbrError AdvertisingProxy::PublishHostAndItsServices(const otSrpServerHost *aHost, OutstandingUpdate *&aUpdate)`? I'm not a fan of using `T *&` as a parameter since it might be error prone.

Reply:
Just found we may not need to return a `OutstandingUpdate`. By the way, we may need to pass in another argument to specify whether we want to generate a new `OutstandinUpdate`.